File: database/datasets-testing/category.py
import csv
import random
import string
import argparse

def random_string(max_length):
    length = random.randint(1, max_length)
    return ''.join(random.choices(string.ascii_letters + string.digits, k=length))

def generate_csv(n):
    with open("./datasets-output/category.csv", mode="w", newline="", encoding="utf-8") as csvfile:
        writer = csv.writer(csvfile)
        writer.writerow(["name", "description"])

        for _ in range(n):
            name = random_string(30)

            # 70% chance de description ser null
            if random.random() < 0.7:
                description = None
            else:
                description = random_string(100)

            writer.writerow([name, description])

if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument("N", type=int, help="Number of entries to generate")

    args = parser.parse_args()
    generate_csv(args.N)

File: api/src/models/category/Category.cs
public class Category {

    public long ID {set; get;}
    public string name {set; get;}
    public string? description {set; get;}

    public Category(long ID, string name) {

        this.ID = ID;
        this.name = name;
        this.description = null;

    }

    public Category(long ID, string name, string? description) {

        this.ID = ID;
        this.name = name;
        this.description = description;

    }

}